websites/D/Discords.com/presence.ts

```diff
@@ -3,351 +3,508 @@ const presence = new Presence({
+				buttons: [{ label: "Apply", url: href }],
+			},
+		},
+		pageNumber = document.querySelector("li.page-item.active")?.textContent,
+		pageNumberBots = document.querySelector("li.active")?.textContent,
+		tagSearch = document
+			.querySelector("h1.text-md-left")
+			?.textContent.replace("Discord Servers", " "),
+		standardSearch = document
+			.querySelector("div.FMDiWzVk")
+			?.querySelector("input")
+			?.getAttribute("value");
+
+	for (const [path, data] of Object.entries(pages))
+		if (pathname.includes(path)) presenceData = { ...presenceData, ...data };
 
-	if (document.location.pathname === "/")
+	if (pathname === "/") {
 		presenceData.details = "Viewing home page";
-	else if (document.location.pathname === "/servers/search") {
+
+		if (standardSearch) {
+			presenceData.details = `🔍 Searching for: ${standardSearch}`;
+			presenceData.buttons = [
+				{
+					label: "View Results",
+					url: href,
+				},
+			];
+		}
+	} else if (pathname === "/servers/search") {
 		presenceData.details = `🔍 Searching for: ${
 			document
 				.querySelector("h1.text-center")
 				?.textContent.replace("Discord Servers", " ") || "Nothing"
 		}`;
-		presenceData.state = `📖 Page ${
-			document.querySelector("li.page-item.active")?.textContent
-		}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
 		presenceData.buttons = [
 			{
 				label: "View Results",
-				url: document.location.href,
+				url: href,
 			},
 		];
-	} else if (document.location.pathname.includes("/servers/tag/")) {
-		presenceData.details = `Viewing 📛 ${
-			document
-				.querySelector("h1.text-center")
-				?.textContent.replace("Discord Servers", " ") || "Nothing"
-		} tag`;
-		presenceData.state = `📖 Page ${
-			document.querySelector("li.page-item.active")?.textContent
-		}`;
+	} else if (pathname.includes("/servers/tag/")) {
+		if (tagSearch) presenceData.details = `🏷️ Viewing tag: ${tagSearch}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
 		presenceData.buttons = [
 			{
 				label: "View Tag",
-				url: document.location.href,
+				url: href,
+			},
+		];
+	} else if (pathname.includes("/servers/tags/")) {
+		if (tagSearch) presenceData.details = `🔑 Viewing keyword: ${tagSearch}`;
+		presenceData.state = `📖 Page ${pageNumber}`;
+		presenceData.buttons = [
+			{
+				label: "View Keyword",
+				url: href,
+			},
+		];
+	} else if (pathname === "/servers/top-100") {
+		presenceData.details = "🏅 Viewing: top-100 servers";
+		presenceData.state = `📖 Page ${pageNumber}`;
+		presenceData.buttons = [
+			{
+				label: "View Top-100",
+				url: href,
+			},
+		];
+	} else if (pathname.includes("/join")) {
+		presenceData.details = "🔗 Joining";
+		presenceData.state =
+			document.querySelector("h2.__9ui9XW8A.mt-4")?.textContent;
+		presenceData.largeImageKey =
+			document.querySelector("div.Xw20bfHH")?.querySelectorAll("img")[1]?.src ??
+			Assets.Logo;
+		presenceData.smallImageKey = Assets.Logo;
+		presenceData.smallImageText = "Discords";
```

Comment:
Small image text isn't really needed as would be displayed as the service name

Reply:
I removed the small image text, but it doesn't seem to default to the service name.

Reply:
Pardon me, I don't think what I said made total sense. What I was meant to say was: the service name would have teh exact same thing so the smallImageText isn't really needed - instead, remove the smallImageText as the service name is unchangable